frontend/app/utils/lookup-utils.server.ts

```diff
@@ -27,3 +27,34 @@ export function localizeAndSortRegions(regions: Region[], locale: string) {
   const sortedRegions = mappedRegions.toSorted((a, b) => a.name.localeCompare(b.name, locale));
   return sortedRegions;
 }
+
+/**
+ * Localizes a single language object by adding a localized name.
+ *
+ * @param language - The language object to localize.
+ * @param locale - The locale code for localization.
+ * @returns The localized language object with a localized name.
+ */
+export function localizeLanguage(language: Language, locale: string) {
+  const { nameEn, nameFr, ...rest } = language;
+  return {
+    ...rest,
+    name: (locale === 'fr' ? nameFr : nameEn) ?? '',
```

Comment:
If it `nameFr` or `nameEn` is `undefined` and let it be. I think we should throw an exception if preferred language names are `undefined`.